### PR TITLE
Bug 1593190 - Provide backend ORM support for ingesting the application name

### PR DESCRIPTION
**Pull request by ionutgoldan**

### CODE REVIEW

File: treeherder/perf/models.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ class PerformanceSignature(models.Model):
     option_collection = models.ForeignKey(OptionCollection, on_delete=models.CASCADE)
     suite = models.CharField(max_length=80)
     test = models.CharField(max_length=80, blank=True)
+    application = models.CharField(max_length=25, null=True,
```
Comment (octavian-negru) [inline treeherder/perf/models.py]: Notice: Please keep the size in sync with [schema](https://phabricator.services.mozilla.com/D51410). I suggested that the length can be smaller.

Comment (ionutgoldan) [reply treeherder/perf/models.py]: Done.

**Comment by ionutgoldan** [comment]
FYI @karlht.

**Comment by camd** [comment]
I am out of town on PTO till November 7th.  Please have @sarah-clements  or @armenzg review these changes if needed.

**Comment by ionutgoldan** [comment]
@armenzg could you schedule this for merge & deploy during this weekend, right after the deploy of Taskcluster-related PRs? It will cause a lengthy deploy, which couldn't be done during next week's working hours.
It is also a dependency for other PRs I have.

**Comment by armenzg** [comment]
@sarah-clements would you be able to merge this after the downtime?
I will check on Sunday to see if it landed if not I will go ahead and do so.

**Comment by sarah-clements** [comment]
> @sarah-clements would you be able to merge this after the downtime?
> I will check on Sunday to see if it landed if not I will go ahead and do so.

Sure, I can do that. However, @ionutgoldan and @octavian-negru please be sure to coordinate when merging your other prs that have backend changes. There are quite a few prs with migrations that all point to the same number `0020...`.

**Comment by sarah-clements** [comment]
I've deployed this to stage today.

**Comment by sarah-clements** [comment]
Deployed to production.

**Comment by ionutgoldan** [comment]
> 
> 
> > @sarah-clements would you be able to merge this after the downtime?
> > I will check on Sunday to see if it landed if not I will go ahead and do so.
> 
> Sure, I can do that - depending on when my part finishes. However, @ionutgoldan and @octavian-negru please be sure to coordinate when merging your other prs that have backend changes. There are quite a few prs with migrations that all point to the same number `0020...`.

Sure thing. Will update all our PRs which point to the same prefix. Thanks!

**Pull request merged**